devtools/gradle/src/main/java/io/quarkus/gradle/QuarkusPlugin.java

```diff
@@ -239,6 +233,31 @@ public void execute(Task test) {
         });
     }
 
+    private void registerConditionalDependencies(Project project) {
+        ConditionalDependenciesEnabler conditionalDependenciesEnabler = new ConditionalDependenciesEnabler(project);
+        ApplicationDeploymentClasspathBuilder deploymentClasspathBuilder = new ApplicationDeploymentClasspathBuilder(
+                project);
+        project.getConfigurations().getByName(JavaPlugin.COMPILE_CLASSPATH_CONFIGURATION_NAME).getIncoming()
+                .beforeResolve((dependencies) -> {
+                    Set<ExtensionDependency> implementationExtensions = conditionalDependenciesEnabler
+                            .declareConditionalDependencies(JavaPlugin.IMPLEMENTATION_CONFIGURATION_NAME);
```

Comment:
In case of the other two configs below, the config name matches but not in this case. We find the COMPILE_CLASSPATH_CONFIGURATION_NAME but looking for the conditional dependencies in IMPLEMENTATION_CONFIGURATION_NAME. Should we be using IMPLEMENTATION_CONFIGURATION_NAME in both cases?

Reply:
Let's not modify the original compileClasspath.